beacon-chain/state/state-native/v3/beacon_state_mainnet.go

```diff
@@ -0,0 +1,54 @@
+// +build !minimal
+
+package v3
+
+import (
+	"sync"
+
+	eth2types "github.com/prysmaticlabs/eth2-types"
+	"github.com/prysmaticlabs/go-bitfield"
+	customtypes "github.com/prysmaticlabs/prysm/beacon-chain/state/state-native/custom-types"
+	"github.com/prysmaticlabs/prysm/beacon-chain/state/state-native/fieldtrie"
+	"github.com/prysmaticlabs/prysm/beacon-chain/state/stateutil"
+	"github.com/prysmaticlabs/prysm/beacon-chain/state/types"
+	ethpb "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1"
+)
+
+// BeaconState defines a struct containing utilities for the eth2 chain state, defining
+// getters and setters for its respective values and helpful functions such as HashTreeRoot().
+type BeaconState struct {
+	genesisTime                  uint64                        `ssz-gen:"true"`
+	genesisValidatorsRoot        customtypes.Byte32            `ssz-gen:"true" ssz-size:"32"`
```

Comment:
We tend to follow the naming conventions `Bytes32` in the rest of the repository, so maybe we can align on consistency in a future PR

Reply:
This will be removed. See https://github.com/prysmaticlabs/prysm/issues/10100